tools/ob_admin/backup_tool/ob_admin_backup_validation_task.cpp

```diff
@@ -0,0 +1,4804 @@
+          STORAGE_LOG(WARN, "result is null", K(ret));
+        } else {
+          if (OB_FAIL(result->next())) {
+            if (OB_ITER_END != ret) {
+              STORAGE_LOG(WARN, "failed to next", K(ret));
+            } else {
+              // no such a deletion tablet
+              STORAGE_LOG(WARN, "missing tablet is not normal deleted", K(ret),
+                          K(missing_tablet_iter->second));
+              int tmp_ret = OB_SUCCESS;
+              char buf[1024];
+              if (OB_TMP_FAIL(databuff_printf(
+                      buf, sizeof(buf) - 1, "Tablet %ld in Backup Set %ld is missing",
+                      missing_tablet_iter->second.id(),
+                      backup_set_attr->backup_set_info_desc_->backup_set_file_.backup_set_id_))) {
+                STORAGE_LOG(WARN, "failed to print buf", K(tmp_ret));
+              } else if (OB_TMP_FAIL(ctx_->go_abort(buf, "missing tablet is not normal deleted"))) {
+                STORAGE_LOG(WARN, "failed to go abort", K(tmp_ret));
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  if (OB_NOT_NULL(ctx_) && OB_FAIL(ret)) {
+    ctx_->go_abort("tenant minor/major data info", "ls minor/major data info failed cross check");
+  }
+  return ret;
+}
+void ObAdminBackupSetMetaValidationTask::post_process_(int ret)
+{
+  int tmp_ret = OB_SUCCESS;
+  if (IS_NOT_INIT) {
+    tmp_ret = OB_NOT_INIT;
+    STORAGE_LOG(WARN, "not init", K(ret), K(tmp_ret));
+  } else if (OB_CANCELED == ret) {
+    STORAGE_LOG(INFO, "backup set meta validation task is canceled", K(ret), K(task_id_));
+  } else if (ctx_->processing_backup_set_id_array_.empty()) {
+    tmp_ret = OB_ITER_END;
+    STORAGE_LOG(INFO, "backup set id is all empty", K(ret), K(task_id_));
+  } else {
+    if (OB_SUCC(ret)) {
+      printf(CLEAR_LINE);
+      int64_t backup_set_id = -1;
+      ObAdminBackupSetValidationAttr *backup_set_attr = nullptr;
+      if (OB_TMP_FAIL(ctx_->processing_backup_set_id_array_.at(task_id_, backup_set_id))) {
+        STORAGE_LOG(WARN, "failed to get backup set id", K(ret), K(task_id_));
+      } else if (OB_TMP_FAIL(ctx_->get_backup_set_attr(backup_set_id, backup_set_attr))) {
+        STORAGE_LOG(WARN, "failed to get backup set store", K(ret), K(backup_set_id));
+      } else if (backup_set_attr->backup_set_info_desc_->backup_set_file_.backup_type_
+                     .is_full_backup()) {
+        printf("Full Backup set %ld passed meta info validation\n", backup_set_id);
+      } else if (backup_set_attr->backup_set_info_desc_->backup_set_file_.backup_type_
+                     .is_inc_backup()) {
+        printf("Inc Backup set %ld passed meta info validation\n", backup_set_id);
+      }
+    } else {
+      printf(CLEAR_LINE);
+      printf("Backup set %ld has corrupted meta info\n",
+             ctx_->processing_backup_set_id_array_.at(task_id_));
+    }
+  }
+  fflush(stdout);
+}
+////////////////ObAdminBackupSetMetaValidationTask End////////////////
```

Comment:
Have we check continues between minor and major?

Reply:
we checked coverage

Reply:
we also checked sstable continuity.